src/transaction/components/TransactionProvider.tsx

```diff
@@ -212,9 +220,23 @@ export function TransactionProvider({
     }
   }, [transactionStatus]);
 
+  // Reset status to pending when executing multiple transactions with EOA
+  useEffect(() => {
+    if (
+      transactionStatus === 'pending' &&
+      lifeCycleStatus.statusName === 'transactionLegacyExecuted'
```

Comment:
shouldn't lifeCycleStatus already have been set by the useEffect above that also checks transactionStatus pending?

Reply:
yes but transaction status doesn't actually change. it is `pending` until it reaches the `success` state so it wasn't getting reset to `transactionPending` after we change it to `transactionLegacyExecuted`